### PR TITLE
MSA-785: SleepSense Automation

### PR DESCRIPTION
**Pull request by SmartThings**
Submitting 'SleepSense Automation' for publication

### CODE REVIEW

**Comment by jasonbio** [comment]
Add comments to the code that describe the main functions, as this SmartApp is on the large side.

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy [hunk 1]
```diff
@@ -0,0 +1,1348 @@
...
+        section(title: "More options", hidden: hideOptionsSection(), hideable: true) {
+            def timeBasedTrigger = trigger in ["At Sunrise", "At Sunset", "At a Specific Time"]
+            log.trace "timeBasedTrigger: $timeBasedTrigger"
+            if (!timeBasedTrigger) {
+                href "timeIntervalInput", title: getLabel("str_TimeSet"), description: timeLabel ?: "Tap to set", state: timeLabel ? "complete" : "incomplete"
+            }
+
+            input "days", "enum", title: getLabel("str_WeekSet"), multiple: true, required: false,
+                options: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
+
```
Comment (jasonbio) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:250]: No Korean translation for day of the week

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy [hunk 2]
```diff
@@ -0,0 +1,1348 @@
...
+    input(name: "sleepSensor", type: "capability.sleepSensor", title: getLabel("str_SelectSense"), multiple: false, submitOnChange: true);
+}
+
+//Sleep sensor action to trigger event
+private actionMap() {
+    state.sleepActionMap = ["sleeping": "When I fall asleep", "not sleeping": "When I wake"];
+    if(sleepSensor.find{it.hasAttribute("bedstate")} != null) {
+        state.sleepActionMap.put("in bed", "When I get in bed");
+        state.sleepActionMap.put("out of bed", "When I get out of bed");
+    }
```
Comment (jasonbio) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:61]: No Korean translations for the triggers

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy [hunk 3]
```diff
@@ -0,0 +1,1348 @@
...
+        }
+    }
+}
+
+def timeIntervalInput() {
+    dynamicPage(name: "timeIntervalInput") {
+        section {
+            input "startTimeType", "enum", title: getLabel("str_TimeStartingCondition"), options: [["time": "A specific time"], ["sunrise": "Sunrise"], ["sunset": "Sunset"]], defaultValue: "time", submitOnChange: true
+            if (startTimeType in ["sunrise","sunset"]) {
+                input "startTimeOffset", "number", title: getLabel("str_TimeOffset"), range: "*..*", required: false
```
Comment (jasonbio) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:261]: No Korean translation for starting condition

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy [hunk 4]
```diff
@@ -0,0 +1,1348 @@
...
+            }
+            else {
+                input "starting", "time", title: getLabel("str_TimeStart"), required: false
+            }
+        }
+        section {
+            input "endTimeType", "enum", title: getLabel("str_TimeEngingCondition"), options: [["time": "A specific time"], ["sunrise": "Sunrise"], ["sunset": "Sunset"]], defaultValue: "time", submitOnChange: true
+            if (endTimeType in ["sunrise","sunset"]) {
+                input "endTimeOffset", "number", title: getLabel("str_TimeOffset"), range: "*..*", required: false
```
Comment (jasonbio) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:270]: No Korean translation for ending condition

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy [hunk 5]
```diff
@@ -0,0 +1,1348 @@
...
+                            break;
+                    }
+                }
+            }
+        }
+    }
+    if(settings.controlType && settings.controlType == "homeWatch") {
+        section(getLabel("str_HomeWatchTitle")) {
+            input(name: "homeWatchAction", type: "enum", title: getLabel("str_HomeWatchSectionSelect"), required: true, multiple: false, options: [off: "Off", stay: "Stay", arm: "Arm"], submitOnChange: true);
+        }
```
Comment (jasonbio) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:221]: No Korean translation

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy [hunk 6]
```diff
@@ -0,0 +1,1348 @@
...
+            }
+        }
+    }
+}
+
+def otherInputs() {
+    if (settings.trigger) {
+        def timeLabel = timeIntervalLabel()
+        section(title: "More options", hidden: hideOptionsSection(), hideable: true) {
+            def timeBasedTrigger = trigger in ["At Sunrise", "At Sunset", "At a Specific Time"]
```
Comment (jasonbio) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:242]: No Korean translation for "More options"

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy [hunk 7]
```diff
@@ -0,0 +1,1348 @@
...
+            "us": "Which sleep sensor?"
+    ]
+    
+    def str_SelectSleepTrigger=[
+            "kr": "트리거를 선택하세요.",
+            "us": "Select trigger."
+    ]
+
+    def str_SelectLocation=[
+            "kr": "허브를 선택하세요.",
```
Comment (jasonbio) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:863]: Not properly translated. Currently translated as 'Please choose hub'.

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy [hunk 8]
```diff
@@ -0,0 +1,1348 @@
...
+            "us": "What would you like to control?"
+    ]
+
+    def str_HubofHomeWatch=[
+            "kr": "홈지키미의 상태",
```
Comment (jasonbio) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:868]: Double check if translation is inline with SHM branding. Jaejun to follow up

**Comment by jasonbio** [comment]
Double check that all preference panes have Korean translations

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy (hunk 1 above)
Comment (jihyouk) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:250]: i will update title, daily

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy (hunk 2 above)
Comment (jihyouk) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:61]: I will update

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy (hunk 3 above)
Comment (jihyouk) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:261]: I will update

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy (hunk 4 above)
Comment (jihyouk) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:270]: I will update

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy (hunk 5 above)
Comment (jihyouk) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:221]: I will update

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy (hunk 6 above)
Comment (jihyouk) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:242]: I will update

File: smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy (hunk 7 above)
Comment (jihyouk) [inline smartapps/smartsolutionsv2/sleeping/smart-sleep-automation.src/smart-sleep-automation.groovy:863]: I will update

**Pull request closed without merging**